pyatlas/package.sh

```diff
@@ -0,0 +1,148 @@
+venv_path="$pyatlas_root_dir/__pyatlas_venv__"
+protofiles_dir="$gradle_project_root_dir/src/main/proto"
+
+
+### abort the script if the pyatlas source folder is not present ###
+####################################################################
+if [ ! -d "$pyatlas_root_dir/$pyatlas_srcdir" ];
+then
+    err_shutdown "pyatlas source folder not found"
+fi
+
+
+### determine if wget is installed ###
+######################################
+if command -v wget ;
+then
+    wget_command="$(command -v wget)"
+else
+    err_shutdown "'command -v wget' returned non-zero exit status: install wget to run this script"
+fi
+
+
+### download protoc and compile the atlas proto files into python ###
+#####################################################################
+echo "Preparing protoc..."
+# hack to grab the protoc version from dependencies.gradle
+protoc_version=$(grep 'protoc' "$gradle_project_root_dir/dependencies.gradle" | awk -F':' '{print $2; exit}' | tr -d "'")
+protoc_path="$pyatlas_root_dir/protoc"
+# detemine what platform we are on
+if [ ! -f "$protoc_path" ];
+then
+    if [ "$(uname)" == "Darwin" ];
+    then
+        download_link="https://repo1.maven.org/maven2/com/google/protobuf/protoc/${protoc_version}/protoc-${protoc_version}-osx-x86_64.exe"
+        "$wget_command" "$download_link" -O "$protoc_path" || err_shutdown "wget of '$download_link' failed"
+    elif [ "$(uname)" == "Linux" ];
+    then
+        download_link="https://repo1.maven.org/maven2/com/google/protobuf/protoc/${protoc_version}/protoc-${protoc_version}-linux-x86_64.exe"
+        "$wget_command" "$download_link" -O "$protoc_path" || err_shutdown "wget of '$download_link' failed"
+    else
+        err_shutdown "unrecognized platform $(uname)"
+fi
+fi
+chmod 700 "$protoc_path"
+
+# complicated mess to handle case where a proto filename has a space
+# basically, 'find' outputs each file separated by a NUL terminator
+# read -r -d '' reads raw input delimited by NUL characters
+while IFS= read -r -d '' protofile
+do
+    "$protoc_path" "$protofile" --proto_path="$protofiles_dir" --python_out="$pyatlas_root_dir/$pyatlas_srcdir/autogen" || err_shutdown "protoc invocation failed"
+done < <(find "$protofiles_dir" -type f -name "*.proto" -print0)
+
+
+### build the module and documentation ###
+##########################################
+# start the venv
+if [ ! -d "$venv_path" ];
+then
+    err_shutdown "missing $venv_path"
+fi
+# shellcheck source=/dev/null
+source "$venv_path/bin/activate"
+
+# copy the LICENSE to the pyatlas folder
+cp "$gradle_project_root_dir/LICENSE" "$pyatlas_root_dir"
+
+# grab the build version from gradle.properties and inject it into setup.py
+atlas_version=$(grep "version=" "$gradle_project_root_dir/gradle.properties" | cut -f2 -d "=")
+# GNU and BSD sed have different "in-place" flag syntax
+if [ "$(uname)" == "Darwin" ];
+then
+    sed -i "" "s/version=.*/version=\"$atlas_version\",/" "$pyatlas_root_dir/setup.py"
```

Comment:
Awesome!

Reply:
A bit of a hack but it works! Maintaining it is my problem anyway :)